lib/pkgcloud/google/storage/client/files.js

```diff
@@ -41,36 +40,26 @@ exports.upload = function (options) {
   var self = this,
     bucket = this._getBucket(options),
     file = this._getFile(bucket, options),
-    uploadOptions = {};
+    metadata = { contentType: options.contentType };
 
   // check for deprecated calling with a callback
   if (typeof arguments[arguments.length - 1] === 'function') {
     self.emit('log::warn', 'storage.upload no longer supports calling with a callback');
   }
 
-  if (options.contentType) {
-    uploadOptions.contentType = options.contentType;
-  } else {
-    uploadOptions.contentType = mime.getType(options.file || options.remote || options);
-  }
-
   var proxyStream = through(),
-    writableStream = file.createWriteStream(uploadOptions);
+    writableStream = file.createWriteStream({ metadata });
 
   // we need a proxy stream so we can always return a file model
   // via the 'success' event
-  writableStream.on('complete', function() {
+  writableStream.on('finish', function() {
     proxyStream.emit('success', new storage.File(self, file));
   });
 
   writableStream.on('error', function(err) {
     proxyStream.emit('error', err);
   });
 
-  writableStream.on('data', function(chunk) {
-    proxyStream.emit('data', chunk);
-  });
-
```

Comment:
Upon a second, deeper look this is _technically_ a breaking change. I cannot recall why we use a proxy stream here. This should be just as valid, could you give it a shot? 

``` js
var writableStream = file.createWriteStream({ metadata });

writableStream.on('finish', function() {
  writableStream.emit('success', new storage.File(self, file));
});

return writableStream;
```

Basically return the raw stream, but still emit the normalized `'success'` event.

Reply:
You're right. Sorry I was to blind to see that the proxy is obsolete. Addressed this in https://github.com/pkgcloud/pkgcloud/pull/666/commits/32eccf5c04738bbe653e84054ee2f7485baff99a